packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -192,27 +404,46 @@ class CupertinoNavigationBar extends StatelessWidget implements ObstructingPrefe
 /// close button in case of a fullscreen dialog) to pop the current route if none
 /// is provided and [automaticallyImplyLeading] is true (true by default).
 ///
+/// The [largeTitle] widget will automatically be a title text from the current
+/// route if none is provided and [automaticallyImplyTitle] is true (true by
+/// default).
+///
+/// When [transitionBetweenRoutes] is true, this navigation bar will transition
+/// on top of the routes instead of inside it if the route being transitioned
+/// to also has a [CupertinoNavigationBar] or a [CupertinoSliverNavigationBar]
+/// with [transitionBetweenRoutes] set to true. If [transitionBetweenRoutes] is
+/// true, none of the [Widget] parameters can contain a key in its subtree since
+/// that widget will exist in multiple places in the tree simultaneously.
+///
 /// See also:
 ///
 ///  * [CupertinoNavigationBar], an iOS navigation bar for use on non-scrolling
 ///    pages.
-class CupertinoSliverNavigationBar extends StatelessWidget {
+class CupertinoSliverNavigationBar extends StatefulWidget {
   /// Creates a navigation bar for scrolling lists.
   ///
   /// The [largeTitle] argument is required and must not be null.
   const CupertinoSliverNavigationBar({
     Key key,
-    @required this.largeTitle,
+    this.largeTitle,
     this.leading,
     this.automaticallyImplyLeading = true,
+    this.automaticallyImplyTitle = true,
+    this.previousPageTitle,
     this.middle,
     this.trailing,
     this.border = _kDefaultNavBarBorder,
     this.backgroundColor = _kDefaultNavBarBackgroundColor,
     this.padding,
     this.actionsForegroundColor = CupertinoColors.activeBlue,
-  }) : assert(largeTitle != null),
-       assert(automaticallyImplyLeading != null),
+    this.transitionBetweenRoutes = true,
+  }) : assert(automaticallyImplyLeading != null),
+       assert(automaticallyImplyTitle != null),
+       assert(
+         automaticallyImplyTitle == true || largeTitle != null,
+         'A largeTitle must be provided. Otherwise, automaticallyImplyTitle '
```

Comment:
The first sentence here seems to be shown to be a lie by the second. :-)

Cosider something like "No largeTitle has been provided but automaticallyImplyTitle is false.\nEither provide a largeTitle or set automaticallyImplyTitle to true."

Reply:
lol, sorry, dumb phrasing. Fixing